cmd/ipfs/daemon.go

```diff
@@ -457,6 +469,99 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 	return errs
 }
 
+const mfsRepinInterval = time.Minute
+
+func pinMFSOnChange(cctx *oldcmds.Context, node *core.IpfsNode) (<-chan error, error) {
+	errCh := make(chan error)
+	go func() {
+		var lastRootCid cid.Cid
+		for {
+			// poll periodically
+			tmo := time.NewTimer(mfsRepinInterval)
+			select {
+			case <-cctx.Context().Done():
+				return
+			case <-tmo.C:
+			}
+
+			// get the most recent MFS root cid
+			rootNode, err := node.FilesRoot.GetDirectory().GetNode()
+			if err != nil {
+				select {
+				case errCh <- err:
+				case <-cctx.Context().Done():
+					return
+				}
+				continue
+			}
+			rootCid := rootNode.Cid()
+
+			// do nothing, if MFS has not changed
+			if lastRootCid == rootCid {
+				continue
+			}
+			lastRootCid = rootCid
+
+			// reread the config, which may have changed in the meantime
+			cfg, err := cctx.GetConfig()
+			if err != nil {
+				select {
+				case errCh <- err:
+				case <-cctx.Context().Done():
+					return
+				}
+			}
+
+			// pin on all remote services in parallel to prevent DoS attacks
+			var wg sync.WaitGroup
+			for svcName, svcConfig := range cfg.Pinning.RemoteServices {
+				if svcConfig.Policies.PinMFS == nil || !*svcConfig.Policies.PinMFS {
+					continue
+				}
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					pinMFS(cctx.Context(), node, lastRootCid, svcName, svcConfig, errCh)
```

Comment:
`go vet` error here on loop closure which are always fun 😄.

> loopclosure: loop variable svcName captured by func literal (govet)

You can either set `name := svcName` in the for loop and use that in the goroutine, or pass arguments into the goroutine function.